core/src/main/java/com/linecorp/armeria/common/util/Sampler.java

```diff
@@ -42,45 +42,52 @@
  *
  * <p>Forked from brave-core 5.6.3 at 3be55b5cccf881104bdd80c93e97d2575b83952d
  */
-// abstract for factory-method support on Java language level 7
-public abstract class Sampler {
-
-    public static final Sampler ALWAYS_SAMPLE = new Sampler() {
-        @Override
-        public boolean isSampled() {
-            return true;
-        }
-
-        @Override public String toString() {
-            return "AlwaysSample";
-        }
-    };
+@FunctionalInterface
+public interface Sampler<T> {
+    /**
+     * Returns a sampler, given a rate expressed as a floating point number between {@code 0.0} and {@code 1.0}.
+     *
+     * @param rate the sampling rate between {@code 0.01} and {@code 1.0}.
```

Comment:
nit: `{@code 0.01}` -> `{@code 0.0}`?

Reply:
Thank you for review. 
I think `{@code 0.01}` is correct in this comment.
Because in the constructor of `CountingSampler` validates with a value of `0.01`.
https://github.com/line/armeria/blob/d42052d7b0d26de4fd23710cb371f73c6d23df60/core/src/main/java/com/linecorp/armeria/common/util/CountingSampler.java#L64

Please let me know if I'm wrong. Thank you! 🙏

Reply:
The above doc says `number between {@code 0.0} and {@code 1.0}`, but it's weird that the parameter description says `between {@code 0.01} and {@code 1.0}`. I think we need to use the same value.

And now I figured why. `0` is a valid value but the value less than `0.01` is not valid.
```java
if (rate == 0) { // 0 is valid.
    return NEVER_SAMPLE;
}
if (rate == 1.0) {
    return ALWAYS_SAMPLE;
}
if (rate < 0.01 || rate > 1) { // What da heck?
    throw new IllegalArgumentException("rate should be between 0.01 and 1: was " + rate);
}
```
So I think we need to fix this.
```java
final class CountingSampler extends Sampler {
    public static Sampler create(final float rate) {
        final int percent = (int) (rate * 100.0f);
        if (percent == 0) {
            return NEVER_SAMPLE;
        }
        if (percent == 100) {
            return ALWAYS_SAMPLE;
        }
        if (percent < 0 || percent > 100) {
            throw new IllegalArgumentException("rate should be between 0.0 and 1: was " + rate);
        }
        return new CountingSampler(rate);
    }

	...
	CountingSampler(int percent, Random random) {
	...
```
What do you think? @geminiKim

Reply:
I think that is good idea, Thank you for review!